src/Murder/Core/Extensions/EnumExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Runtime.CompilerServices;
+
+namespace Murder.Core.Extensions;
+
+internal static class EnumExtensions
+{
+    /// <summary>
+    /// Implementation of .HasFlag which does not box the Enum and thus does not allocate. Useful in parts
+    /// where a `HasFlag` check needs to happen every frame.
+    /// </summary>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static bool NonAllocatingHasFlag<TEnum>(this TEnum lhs, TEnum rhs) where TEnum : unmanaged, Enum
```

Comment:
I would add a big comment here specifying when this would be useful (re: debug builds or super long methods). I don't _think_ any of the methods would currently fall into that? But I'm still okay that they call that, since I don't know for sure what "super long" means.